test/common/selfmanaged_apicast.go

```diff
@@ -0,0 +1,889 @@
+	"net/http"
+	"os"
+	"os/exec"
+	ctrl "sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	apicastOperatorVersion       = "0.5.0"
+	apicastImageStreamTag        = "3scale2.11.0"
+	apicastNamespace             = "selfmanaged-apicast"
+	apiExampleApicast            = "apicast-example-apicast"
+	adminPortalCredentialsSecret = "adminportal-credentials"
+	namespacePrefix              = "redhat-rhoam-"
+	//defaultDedicatedAdminName    = "customer-admin"
+	accountOrgName               = "Developer"
+	serviceSystemName            = "api"
+)
+
+var (
+	log                 = l.NewLogger()
+	threeScaleNamespace string
+)
+
+func TestSelfmanagedApicast(t TestingTB, testingCtx *TestingContext) {
+	threeScaleNamespace = fmt.Sprintf("%s3scale", namespacePrefix)
+	scheme := runtime.NewScheme()
+	utilruntime.Must(operatorsv1alpha1.AddToScheme(scheme))
+	utilruntime.Must(v1.AddToScheme(scheme))
+	utilruntime.Must(routev1.AddToScheme(scheme))
+	utilruntime.Must(appsv1alpha1.SchemeBuilder.AddToScheme(scheme))
+
+	config := ctrl.GetConfigOrDie()
+	client, err := k8sclient.New(config, k8sclient.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		t.Fatalf("Error get Client k8sclient: %v", err)
+	}
+	ctx := context.Background()
+
+	err = cleanUpBeforeTest(ctx, client)
+	if err != nil {
+		t.Fatalf("Error clean before test: %v", err)
+	}
+
+	// Create apicast namespace
+	err = createApicastNamespace(ctx, client)
+	if err != nil {
+		t.Fatalf("Error create namespace  %s: %v", apicastNamespace, err)
+	}
+
+	// Get 3scale admin token
+	token3scale, err := getThreeScaleAdminToken(ctx, client)
+	if err != nil {
+		t.Fatalf("Error get 3scale admin Token: %v", err)
+	}
+
+	// Customer login
+	err = customerLogin()
+	if err != nil {
+		t.Fatalf("Error in customer login: %v", err)
+	}
+
+	// Import APIcast Image
+	err = importApicastImage()
+	if err != nil {
+		t.Fatalf("Error import Apicast image: %v", err)
+	}
+
+	// Create an adminportal-credentials secret
+	threeScaleAdminPortal, err := getThreeScaleAdminPortal(ctx, client)
+	if err != nil {
+		t.Fatalf("Error get 3scale admin portal: %v", err)
+	}
+	err = createAdminPortalCredentialsSecret(ctx, client, token3scale, threeScaleAdminPortal)
+	if err != nil {
+		t.Fatalf("Error create adminportal-credentials secret: %v", err)
+	}
+
+	// Use self-managed APIcast instead of the builded one for API
+	userKey, err := promoteSelfManagedAPIcast(ctx, client, threeScaleAdminPortal, token3scale)
+	if err != nil {
+		t.Fatalf("Error promote Self-managed APIcast: %v", err)
+	}
+
+	// Install 3scale APIcast gateway operator
+	err = installThreeScaleApicastGatewayOperator(client)
+	if err != nil {
+		t.Fatalf("Error install 3scale APIcast gateway Operator: %v", err)
+	}
+
+	// Create a self-managed APIcast
+	err = createSelfManagedApicast(ctx, client)
+	if err != nil {
+		t.Fatalf("Error create self-managed APIcast: %v", err)
+	}
```

Comment:
Still seeing CRD not found error on fresh install / run :thinking: 
```
xml"
time="2022-01-06T11:27:19Z" level=info msg="userKey found"
time="2022-01-06T11:27:19Z" level=info msg="Install 3scale APIcast gateway Operator"
time="2022-01-06T11:27:25Z" level=info msg="Install plan found for subscription apicast-operator"
time="2022-01-06T11:27:25Z" level=info msg="Create self-managed APIcast"
I0106 11:27:26.193441   95543 request.go:655] Throttling request took 1.000415585s, request: GET:https://api.chfan.je0y.s1.devshift.org:6443/apis/helm.openshift.io/v1beta1?timeout=32s
time="2022-01-06T11:27:28Z" level=error msg="Error create APIcast - example-apicast" error="no matches for kind \"APIcast\" in version \"apps.3scale.net/v1alpha1\""
```

Reply:
Yes, I saw it once. Seems as before when trying first time... I will try look at code again, but it seems it's waiting . When running second and more times I saw that it's waiting in "Create self-managed APIcast" and it always succeded. But Yes, first time I saw it, and actually wanted to tell you